simple/rdm_tagged_peek.c

```diff
@@ -31,95 +31,187 @@
 #include <stdlib.h>
 #include <string.h>
 #include <getopt.h>
+#include <unistd.h>
 
 #include <rdma/fi_errno.h>
 #include <rdma/fi_tagged.h>
 
 #include <shared.h>
 
-static int tagged_peek(uint64_t tag)
+static int wait_for_send_comp(int count)
 {
+	int ret, completions = 0;
 	struct fi_cq_tagged_entry comp;
-	struct fi_msg_tagged msg;
+
+	do {
+		ret = fi_cq_sread(txcq, &comp, 1, NULL, -1);
```

Comment:
I propose to update it as follows:
`ret = fi_cq_sread(txcq, &comp, 1, NULL, -1);
 if ((ret < 0) && (ret != -FI_EAGAIN)) {
 		 FT_PRINTERR("fi_cq_sread", ret);
 		 return ret;
 }`

if `fi_cq_sread()` returns _-FI_EAGAIN_, it means that right now there are no events and we need to spin one more time to wait new events.